File: Editorial/0042_Trapping_Rain_Water/Go/4_Using_2_pointers.go
func trap(height []int) int {
    left := 0
    right := len(height) - 1
    ans := 0
    left_max := 0
    right_max := 0
    for left < right {
        if height[left] < height[right] {
            if height[left] >= left_max {
                left_max = height[left]
            } else {
                ans += left_max - height[left]
            }
            left++
        } else {
            if height[right] >= right_max {
                right_max = height[right]
            } else {
                ans += right_max - height[right]
            }
            right--
        }
    }
    return ans
}

File: Editorial/42_Trapping_Rain_Water/C/4_Using_2_pointers.c
int trap(int* height, int heightSize) {
    int left = 0, right = heightSize - 1;
    int ans = 0;
    int left_max = 0, right_max = 0;
    while (left < right) {
        if (height[left] < height[right]) {
            height[left] >= left_max ? (left_max = height[left])
                                     : (ans += (left_max - height[left]));
            ++left;
        } else {
            height[right] >= right_max ? (right_max = height[right])
                                       : (ans += (right_max - height[right]));
            --right;
        }
    }
    return ans;
}
